packages/lisk-p2p/src/p2p.ts

```diff
@@ -494,6 +511,44 @@ export class P2P extends EventEmitter {
 		return seedPeerUpdatedInfos;
 	}
 
+	private _pickRandomDiscoveredPeers(
+		count: number,
+	): ReadonlyArray<P2PDiscoveredPeerInfo> {
+		const discoveredPeerList: ReadonlyArray<P2PDiscoveredPeerInfo> = [
+			...this._newPeers.values(),
+			...this._triedPeers.values(),
+		]; // Peers whose values has been updated atleast once.
+
+		return selectRandomPeerSample(discoveredPeerList, count);
+	}
+
+	private _handleGetPeersRequest(request: P2PRequest): void {
+		// TODO later: Remove fields that are specific to the current Lisk protocol.
+		const peers = this._pickRandomDiscoveredPeers(MAX_PEER_LIST_BATCH_SIZE).map(
+			(peerInfo: P2PDiscoveredPeerInfo): ProtocolPeerInfo => {
+				const { ipAddress, ...peerInfoWithoutIp } = peerInfo;
+
+				// The options property is not read by the current legacy protocol but it should be added anyway for future compatibility.
+				return {
+					...peerInfoWithoutIp,
+					ip: ipAddress,
+					broadhash: peerInfoWithoutIp.broadhash
+						? (peerInfoWithoutIp.broadhash as string)
+						: '',
+					nonce: peerInfoWithoutIp.nonce
```

Comment:
Is it necessary to default `broadhash` and `nonce` to empty strings if not provided?

Reply:
Actually, `ProtocolPeerInfo` has broadhash and nonce as required field so I need to default  it to accept because its unknown type in P2PDiscoveredPeer.

Reply:
So either we can allow it to accept `undefined` also or else we provide a default value,  if `undefined.`